js/page-specific/site-search-results.js

```diff
@@ -0,0 +1,186 @@
+namespacer('bcpl.pageSpecific');
+
+bcpl.pageSpecific.swiftypeSearchResults = (($, querystringer, Handlebars, constants) => {
+	let $searchResultsTarget;
+	const searchResultsTargetSelector = '#search-results-target';
+	const templateSelector = '#swiftype-search-results-template';
+	const errorMessageHtml = '<div class="main-content top-border"><p>There were no results found for this search.</p></div>';
+
+	const cleanSearchTerm = (termToClean) => {
+		let cleanedSearchTerm = termToClean.trim();
+		const safeSearchTerms = [];
+		// Ensure the last character is not '+' as the the trailing space causes no results
+		if (cleanedSearchTerm[cleanedSearchTerm.length - 1] === '+') {
+			cleanedSearchTerm = cleanedSearchTerm.slice(0, -1);
+		}
+
+		const searchTerms = cleanedSearchTerm.split('+');
+		for (let i = 0, len = searchTerms.length; i < len; i += 1) {
+			let term = decodeURIComponent(searchTerms[i]);
+			term = term
+				.replace(/[?#\\/]/g, ' ')
+				.replace(/&/g, 'and').trim();
+
+			const encodedTerm = encodeURIComponent(term);
+			safeSearchTerms.push(encodedTerm);
+		}
+		return safeSearchTerms.join('%20');
+	};
+
+	const getSearchResults = (searchTerm, pageNumber) => {
+		const currentPageNumber = pageNumber || 1;
+		const cleanedSearchTerm = cleanSearchTerm(searchTerm);
+		const requestUrl = `${constants.baseApiUrl}${constants.search.urls.api}/${cleanedSearchTerm}/${currentPageNumber}`;
+
+		$.ajax(requestUrl)
+			.then(searchResultRequestSuccessHandler, searchResultRequestErrorHandler);
+	};
+
+	const postClickThroughData = (searchTerm, id, destinationUrl) => {
+		const cleanedSearchTerm = cleanSearchTerm(searchTerm);
+		const requestUrl = `${constants.baseApiUrl}${constants.search.urls.trackClickThrough}/${cleanedSearchTerm}/${id}`;
+
+		$.ajax({
+			type: 'POST',
+			url: requestUrl
+		})
+			.then(() => postClickThroughSuccess(destinationUrl), searchResultRequestErrorHandler);
+	};
+
+	const postClickThroughSuccess = (destinationUrl) => {
+		window.location = destinationUrl;
+	};
+
+	const searchResultRequestErrorHandler = (err) => {
+		console.error(err); // eslint-disable-line no-console
+	};
+
+	const buildResultSettings = (result) => {
+		const highlight = result.highlight.body || result.highlight.sections || result.highlight.title;
+		const title = result.title;
+		const url = result.url;
+		const id = result.id;
+
+		return {
+			highlight,
+			title: title,
+			url: url,
+			id: id
+		};
+	};
+
+	const buildSearchResults = (hits) => {
+		const results = [];
+
+		for (let i = 0, hitCount = hits.length; i < hitCount; i += 1) {
+			results.push(buildResultSettings(hits[i]));
+		}
+
+		return results;
+	};
+
+	const buildPageLinks = (lastPage, currentPage) => {
+		const pageLinks = [];
```

Comment:
same as above, think you can simplify with map.

Reply:
No can do -- I'm not iterating over a collection.